src/com/reco1l/osu/hud/HUDSkinData.kt

```diff
@@ -0,0 +1,110 @@
+package com.reco1l.osu.hud
+
+import com.reco1l.andengine.Anchor
+import com.reco1l.framework.math.Vec2
+import com.reco1l.osu.hud.elements.HUDAccuracyCounter
+import com.reco1l.osu.hud.elements.HUDComboCounter
+import com.reco1l.osu.hud.elements.HUDHealthBar
+import com.reco1l.osu.hud.elements.HUDPieSongProgress
+import com.reco1l.osu.hud.elements.HUDScoreCounter
+import com.reco1l.toolkt.data.putObject
+import org.json.JSONArray
+import kotlin.reflect.KClass
+
+data class HUDSkinData(val elements: List<HUDElementSkinData>) {
+
+
+    fun hasElement(type: Class<out HUDElement>) = elements.any { it.type == type }
+
+
+    companion object {
+
+        /**
+         * The default layout data for the HUD.
+         * Based on the default skin layout from osu!stable.
+         */
+        @JvmField
+        val Default = HUDSkinData(
+            listOf(
+                HUDElementSkinData(type = HUDAccuracyCounter::class),
+                HUDElementSkinData(type = HUDComboCounter::class),
+                HUDElementSkinData(type = HUDPieSongProgress::class),
+                HUDElementSkinData(type = HUDHealthBar::class),
+                HUDElementSkinData(type = HUDScoreCounter::class)
+            )
+        )
+
+
+        @JvmStatic
+        fun writeToJSON(data: HUDSkinData) = JSONArray().apply {
+            data.elements.forEach {
+                putObject {
+                    put("type", it.type.simpleName)
```

Comment:
Relying on class names is *very* sensitive to refactors - something that I would not recommend for maintainability. Despite being potentially easy to identify by players, there needs to be another way to identify the type of stored data.